### PR TITLE
Remove WebTools shim from 2022 extension

### PR DESCRIPTION
**Pull request by MariaSolOs**
This PR removes the WebTools synchronous shim from the 2022 extension and instead updates the Angular component content type to extend from the HTML LSP content type, allowing us to also provide completions inside `{{ }}`:
Screenshot 2022-10-26 101255

This only affects the 2022 version of the extension, since in 2019 we get no completions at all without the WebTools shim.

### CODE REVIEW

**Comment by PranavSenthilnathan** [comment]
I remember you mentioned that HTML LSP was a preview feature. Does this work even without that? Also wondering if it works on 17.0?

**Comment by MariaSolOs** [comment]
@PranavSenthilnathan yeah this is the result with the HTML LSP editor disabled. If we enable it, the content type changes so in fact the extension is disabled in non-TS files in that case.
However now that you mention it there's no reason not to add the content type to the client so that it also works with the HTML LSP editor enabled.

**Comment by MariaSolOs** [comment]
**Sigh** this doesn't work in 17.0... :(

File: AngularLanguageService.Shared/LanguageServer/AngularConstants.cs [hunk 1]
```diff
@@ -0,0 +1,11 @@
+﻿namespace AngularLanguageService.Shared.LanguageServer
+{
+    internal static class AngularConstants
+    {
+        internal const string TypeScriptContentTypeName = "TypeScript";
+
+        internal const string AngularComponentContentTypeName = "angularcomponent";
+        internal const string AngularComponentDelegationContentTypeName = "angularcomponent-delegation";
```
Comment (jimmylewis) [inline AngularLanguageService.Shared/LanguageServer/AngularConstants.cs:11]: nit: the -delegation postfix is a detail of the HTML LSP implementation in VS.  It's not really relevant to the Angular implementation.  If the content types can be consolidated (see other comment), this wouldn't be necessary.

File: AngularLanguageService.Shared/LanguageServer/AngularComponentContentTypeDefinitions.cs [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+namespace AngularLanguageService.Shared.LanguageServer
+{
+    internal static class AngularComponentContentTypeDefinitions
+    {
+#pragma warning disable CS0649 // Field 'field' is never assigned to, and will always have its default value null.
+        [Export]
+        [Name(AngularConstants.AngularComponentContentTypeName)]
+        [BaseDefinition(HtmlContentTypeDefinition.HtmlContentType)]
+        [BaseDefinition(CodeRemoteContentDefinition.CodeRemoteContentTypeName)]
+        internal static ContentTypeDefinition AngularComponentContentType;
```
[the remark this answers is not in the document]
Comment (MariaSolOs) [reply AngularLanguageService.Shared/LanguageServer/AngularComponentContentTypeDefinitions.cs:18]: Yeah I had 2 different types for each version of the extension, but using compilation symbols sounds cleaner.

File: AngularLanguageService.Shared/LanguageServer/AngularConstants.cs (hunk 1 above)
Comment (MariaSolOs) [reply AngularLanguageService.Shared/LanguageServer/AngularConstants.cs:11]: I was just using the `-delegation` postfix to be clear about the different between the content types (which relies on one of them inheriting from `html-delegation`) and so I was just following that "convention". I'll follow your recommendation of using `#if` directives though.

**Review by PranavSenthilnathan — APPROVED** [review]

**Pull request merged**